### ISSUE TITLE
Mapped drives // unc path?

### ISSUE THREAD

**Issue by miglety** [ALonelyJuicebox/OFMetadataToStash#6]
Trying to run this in a windows vm that runs DC's script and have tried mapping my stash directory and my OF directory but each time i try to run your script it tells me to define a valid filepath for stash or OF. If i move the stash db local it tells me i need to define a valid filepath for OF data. If i move the user_data.db for one person local and run it the script runs but then tells me that its referencing data thats not in my stash db.

Is there anyway for me to run this in a vm where all of my data is on another machine?

**Comment by ALonelyJuicebox** [comment]
Should work now using either "low" or "normal" metadata match modes (as of 0.4)

**Comment by miglety** [comment]
Just tested the new version and i get this:

`At C:\Users\miglety\Desktop\OFMetadataToStash-0.4\OFMetadataToStash.ps1:270 char:25
+                         if($performername && $OFDBMedia.size){
+                                           ~~
The token '&&' is not a valid statement separator in this version.
    + CategoryInfo          : ParserError: (:) [], ParseException
    + FullyQualifiedErrorId : InvalidEndOfLine`

Also, all of my user_dbs are in folders named after the username vs one folder with all of them in it. I assume i will just need to update the config to point to one user at a time or will this crawl through all of the directories?

**Comment by ALonelyJuicebox** [comment]
> Just tested the new version and i get this:
> 
> ```
> At C:\Users\miglety\Desktop\OFMetadataToStash-0.4\OFMetadataToStash.ps1:270 char:25
> +                         if($performername && $OFDBMedia.size){
> +                                           ~~
> The token '&&' is not a valid statement separator in this version.
>     + CategoryInfo          : ParserError: (:) [], ParseException
>     + FullyQualifiedErrorId : InvalidEndOfLine
> ```
> 

As far as I can tell, the code mentioned in the error you posted does not exist in the 0.4 release. 
Highly suggest that you download the release zip to ensure you have all the right files. 
https://github.com/ALonelyJuicebox/OFMetadataToStash/releases/tag/v0.4

> Also, all of my user_dbs are in folders named after the username vs one folder with all of them in it. I assume i will just need to update the config to point to one user at a time or will this crawl through all of the directories?

You have the option for all of the above! 🙂
You can point the config at the overall OnlyFans directory and have it find all your db files (which will then allow you to either tell it to parse all of them or parse a specific one) or you can specifically tell the config which db file to work with. 

Be aware that the config file has changed in this release and the script will now help you create it on first run. No need to edit the config file directly unless you want to.

**Comment by ALonelyJuicebox** [comment]
Sorry for the hassle, just now realizing I messed up the release a bit-- while the file in the body of the release was accurate, the "source code" zip was not. Should be fixed now!
https://github.com/ALonelyJuicebox/OFMetadataToStash/releases/tag/v0.4a

**Comment by miglety** [comment]
Is there any sort of logging that happens so i can see if its doing anything? Had it running now for about an hour and its still sitting at scanning for existing onlyfans metadata

**Comment by ALonelyJuicebox** [comment]
If you point it the script at a folder with a ton of subfolders to parse, the `get-childitem -recurse` command that's running behind the scenes will probably take longer than you'd like. Can you try pointing the script at a folder that's more specific and see how it performs?

Ex. Point the script at `C:\Media\OnlyFans\Jane\` or `C:\Media\Onlyfans` instead of `C:\`

**Comment by miglety** [comment]
New issue - im on stash .18 and when i run the script it tells me my db is outdated. I did the whole NEver tell me the odds thing and then it runs and then gives me this error. I have verified the pssqlite module is imported as well

```
Invoke-SqliteQuery : Exception calling "Open" with "0" argument(s): "unable to open database file"
At C:\Users\miglety\Desktop\OFMetadataToStash 0.4\OFMetadataToStash.ps1:576 char:44
+ ... eryResult = Invoke-SqliteQuery -Query $Query -DataSource $PathToStash ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Invoke-SqliteQuery
```

**Comment by ALonelyJuicebox** [comment]
Is Stash running at the same time as the script? It is curious that the script cannot open Stash's database file. It's also curious that you are on Stash 18 and you have a different database schema.

Can you give me your build hash? In Stash, click on the gear icon on the upper right to enter Settings then click on About 

For example, for my build of Stash I'm running...
v0.18.0
Build hash:
8a649f02

**Comment by miglety** [comment]
Yes stash was running, will stop it and try again. In the meantime here is my build hash

v0.18.0
Build hash:
8a649f02

**Comment by miglety** [comment]
I did a scan then stopped Stash and ran it again and the same unable to open error happened as well as the out of date error.

**Comment by ALonelyJuicebox** [comment]
Gotcha, okay one last question. If you open Powershell and you run `$PSVersionTable.PSVersion` what does powershell say back?
I can replicate what you're seeing if I run my script on Powershell version 5 (the version that comes with Windows 10) but it seems to run just fine on Powershell version 7.

I'll do some troubleshooting to see what specifically Powershell 5 seems to fall down on, but in the interim, an easy fix should be to just install Powershell 7 

**Direct installation Link**
https://github.com/PowerShell/PowerShell/releases/download/v7.3.0/PowerShell-7.3.0-win-x64.msi

**(or here's the link to Microsoft's website if you want some reading material)**
https://learn.microsoft.com/en-us/powershell/scripting/install/installing-powershell-on-windows?view=powershell-7.3

**Comment by miglety** [comment]
Still having the database is out of date error. 

updating to powershell 7 got rid of some of the other issues, below is what im seeing now.

```
Invoke-SqliteQuery: C:\Users\miglety\Desktop\OFMetadataToStash 0.4\OFMetadataToStash.ps1:576
Line |
 576 |  … eryResult = Invoke-SqliteQuery -Query $Query -DataSource $PathToStash …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "Open" with "0" argument(s): "unable to open database file"
```
And even though ive done a fresh scan, and can find these files in stash, it says this(i changed the config to just look in one directory): 

```
- Missing Files -
There is available metadata for 566 files in your OnlyFans Database that cannot be found in your Stash Database.
    - Be sure to review the MissingFiles log.
    - There's a good chance you may need to rescan your OnlyFans folder in Stash and/or redownload those files

****** Import Complete ******
- Modified Scenes/Images: 0
- Scenes/Images that already had metadata: 0
- This script took 37.0399302 seconds to execute

```

**Comment by ALonelyJuicebox** [comment]
Delete the config file and try again now that you've installed PS7. I'm curious if the path to your Stash db was configured such that it's using forward slashes rather than backwards ones (as is typical for Windows paths)

**Comment by miglety** [comment]
Looks like that fixed it. last question, is there a way for it to just go through all models and create vs selecting them one by one? Lets say you have 100 models, that would be fairly time consuming to run it, let it scan, then pick 1, let it finish and repeat.

**Comment by ALonelyJuicebox** [comment]
Glad to hear it! I'll see if I can throw some guardrails in place to help prevent folks from running into similar hurdles on the next release. Thanks for the feedback!

**Comment by miglety** [comment]
Running it across all models now and im seeing this pop up every now and then, is this something to be concerned about?

```
Invoke-SqliteQuery: C:\Users\miglety\Desktop\OFMetadataToStash 0.4\OFMetadataToStash.ps1:679
Line |
 679 |  … ateResult = Invoke-SqliteQuery -Query $Query -DataSource $PathToStash …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "Fill" with "1" argument(s): "SQL logic error near "11683": syntax error"
Invoke-SqliteQuery: C:\Users\miglety\Desktop\OFMetadataToStash 0.4\OFMetadataToStash.ps1:683
Line |
 683 |  …             Invoke-SqliteQuery -Query $Query -DataSource $PathToStash …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Exception calling "Fill" with "1" argument(s): "SQL logic error near "11683": syntax error"
```

**Comment by ALonelyJuicebox** [comment]
Interesting, not quite sure what's going on there but I'll try to replicate.

I wouldn't be concerned from a db corruption perspective. That query is attempting to associate an image to a performer. In an erroneous scenario as you're seeing here, it'll just fail to add the performer to the image, there's no hard landing. My initial guess is that for whatever reason, the image id or the performer id is missing from the query.

**Comment by miglety** [comment]
Random question, is there a way to get your script to auto set a performer image randomly as it adds performers too?

**Comment by ALonelyJuicebox** [comment]
Hoping I can get that working for the 0.5 release actually. More than likely it would pull from `<performername>\Profile\Avatars` rather than a random image (though that is also possible). We'll see!

**Comment by ALonelyJuicebox** [comment]
Sorry for the delayed response. Yes, if you don't have a user_data.db file, you can still have the script parse based on filepath. It'll create the performer and everything as well. On the main menu, select the second option.

**Comment by miglety** [comment]
That doesnt seem to do anything for me. I tried mapping the directory to multiple specific models that didnt have a user_data.db and each type it would just end immediately after without creating anything. 

```
Press [Enter] to begin:


****** Import Complete ******
- Modified Scenes/Images: 0
- Scenes/Images that already had metadata: 0
- This script took 1.7220721 seconds to execute
```

**Comment by ALonelyJuicebox** [comment]
What do your paths look like? You can obfuscate the actual performer name/filename.
Example: `T:\Onlyfans\LoremIpsum\Posts\Free\Videos\0kjdnasfgkljdsafg_source.mp4`

Second question, where is your OnlyFans data and how are you accessing it from your Windows machine? A mapped drive? A network location? You mentioned your content is on another machine earlier.

**Comment by miglety** [comment]
Using UNC paths in the config file

Some of the paths would be like: \\ip\Personal\OF2\OnlyFans\model\

some contain subfolders of picture and video. Some contain subfolders of Messages, Posts, Profile -

**Comment by ALonelyJuicebox** [comment]
That's pretty strange. For clarity, here's how the script _basically_ works.
The script in this mode will align media with a performer by matching the media's filepath on your filesystem (as seen by the script) with the filepaths in your Stash database.

The script is a bit sensitive in this way, which is why I have that disclaimer before the scan starts telling folks to "run this script on the same machine as Stash"-- it's an attempt to mitigate a situation where the filepath the script sees is different from what's in Stash. 

Open up a media file in Windows Explorer and observe the file path.
Then, with Stash running, find that same Scene and click on the File Info tab in order to view the path as Stash sees it. 

Are these two paths the same?
Second question-- having run the script, are there any new performers created in your Stash correlating to the OF performers you scanned?

**Comment by miglety** [comment]
No it is not the same path. Im running stash in a docker container and have the \Personal\OF2\OnlyFans\ mapped to \of. So stash sees it as: file:///of/OnlyFans/model/video/0glktf9ahc6etyj8e47k1_source.mp4 whereas explorer would see it as: \ip\Personal\OF2\OnlyFans\model\.

When running the script with option 2, nothing is added for any performer. Running option 1 did add 90% of my performers.

**Comment by ALonelyJuicebox** [comment]
Yeah that'll do it... The script (in this mode) is compiling a list of filepaths as it sees them, looking in Stash DB for those same paths, finding no matching paths, and calling it a day.

I have some ideas for (potentially) solving this in the future. Stay tuned-- thanks for the feedback. 🙏🏿

**Comment by twist3dimages** [comment]
Run the script (OFMetatadataToStash.ps1) in the root directory of your Onlyfans folder. Setup your config to just use ".sites" so it matches what your user_data.db says

**Comment by ALonelyJuicebox** [comment]
Should be addressed in 0.6, please run search specificity mode to low
